### PR TITLE
[Bromley] Report missed bin only within two working days of collection.

### PR DESCRIPTION
**Pull request by dracos**
[skip changelog]
Fixes https://github.com/mysociety/fixmystreet-commercial/issues/1893

### CODE REVIEW

File: perllib/FixMyStreet/Cobrand/Bromley.pm [hunk 1]
```diff
@@ -546,4 +547,17 @@ sub bin_services_for_address {
     return \@out;
 }
 
+sub report_allowed {
+    my $dt = shift->clone;
```
Comment (struan) [inline perllib/FixMyStreet/Cobrand/Bromley.pm:551]: Small thing but either a comment to explain what's happening here or different names for variables that make it a bit more obvious would be nice

File: t/app/controller/hercules.t [hunk 2]
```diff
@@ -44,14 +50,23 @@ FixMyStreet::override_config {
...
         $mech->get_ok('/hercules');
         $mech->submit_form_ok({ with_fields => { postcode => 'BR1 1AA' } });
         $mech->submit_form_ok({ with_fields => { address => '1000000002' } });
         $mech->content_contains('2 Example Street');
         $mech->content_contains('Food waste collection');
     };
     subtest 'Report a missed bin' => sub {
+        $mech->content_contains('service-101');
+        $mech->content_contains('service-102');
+        $mech->content_lacks('service-535');
```
Comment (struan) [inline t/app/controller/hercules.t:63]: possibly some test labelling here?

File: perllib/FixMyStreet/Cobrand/Bromley.pm (hunk 1 above)
Comment (dracos) [reply perllib/FixMyStreet/Cobrand/Bromley.pm:551]: Actually I should be using the functions Zurich already have!

**Comment by dracos** [comment]
Bit changed, after realising Zurich already had a working day calculator I've now factored out.

File: perllib/FixMyStreet/Cobrand/UK.pm [hunk 3]
```diff
@@ -449,4 +452,50 @@ sub check_recaptcha {
...
+sub public_holidays {
+    my $nation = shift || 'england-and-wales';
+    my $json = _get_bank_holiday_json();
+    return [ map { $_->{date} } @{$json->{$nation}{events}} ];
+}
+
+sub is_public_holiday {
+    my %args = @_;
+    $args{date} ||= localtime;
+    $args{date} = $args{date}->date;
```
Comment (struan) [inline perllib/FixMyStreet/Cobrand/UK.pm:464]: Does localtime return an object?

Comment (dracos) [reply perllib/FixMyStreet/Cobrand/UK.pm:464]: Yes, if you "use Time::Piece" :)
(The commits are re-ordered, I think github displays them in the wrong order though.)

**Pull request merged**